dataset_interfaces/interface.py

```diff
@@ -445,15 +462,13 @@ def _proxy_cost_callback(self, cost_usd: float) -> None:
     def evaluate_correct(self, *args):
         raise NotImplementedError("This method should not be called. Each test example has its own evaluation function.")
 
-    @abstractmethod
-    def answer_statement_idx(self, example: TestExample) -> Tuple[int, int]:
-        pass
-
     def create_example(self, **kwargs) -> DynamicExample:
+        if self.memory_span > 0:
+            kwargs["memory_span"] = self.memory_span
```

Comment:
Shouldn't this be always passed on to the test example? (Or actually not needed, if the example takes it automatically from the dataset generator).

Reply:
This is one of those really messy situation arising from too much coupling between `TestExample` and `DatasetInterface`. They both depend on each other in different places. `DatasetInterface` creates `TestExample` in `create_examples`, and `TestExample` creates `DatasetInterface` in `TestExample.load()`.

This little bit here was because the `load()` path creates the `DatasetInterface` instance without a configuration, then `instance.memory_span` is 0. So I had to save that value as part of the `TestExample`.

A nicer way to do this could be in injecting the `DatasetInterface` instance to the `load()` method, so we can instantiate it correctly outside the `load()` and decouple it a bit from `TestExample`.